README.md

```diff
@@ -1,80 +1,190 @@
 
-   Hint: It might be useful to have a look in **accountsConfig.js** after you have acquired the packages.
+#### 4. Profiles
+##### Basic information
+Now that users can sign in, they should be able to do something with their own profiles. As you can see, we have chosen to not hand out names for our users this time. However, this is something you should change. Open *profile.html* and create fields for the users profile. These should include firstname, lastname, adress and phone. Since it looks kind of weird when you first sign in and don't have a name, feel free to use the function we provided in the [chat-application](https://github.com/EirikV/meteor-workshop/tree/chat-solution) to hand out a temporary name as users sign in.
 
-   So why does Meteor need a new packaging system? Why can't they just use npm or something?
-   According to Meteor, this is because Meteor applications, and thereby packages have the ability to be isomorphic.
+*Tip: Remember to subscribe to profiles (which also need to be published) through a *waitOn* call in *router.js*. Profiles should be available in all routes, so this should be done globally. Publishing from the *Meteor.users* collection is a little different than other collections. You have to add the fields 'username' and 'profile' like this: `Meteor.users.find({},{fields: 'username': 1, 'profile': 1}});`*
 
-2. Once you've got accounts up and running, try to add some inputs to the profile page. This could, for example be, first- and lastname, address, and phone number.
-At this point it would be awesome to test out template inclusion (`{{>profileFormField}}`).
+*Tip: There are several ways to set a temporary name for a user on login. One is to use the [accountsServer.onCreateUser](http://docs.meteor.com/#/full/accounts_oncreateuser) event. Another, which is what we have done in the solution, is to utilize a helper that returns a name if present, and creates one through a *Meteor.method* if not.*
 
-   Templates in Meteor look like mustache or handlebars, but are not.
-   Meteor uses its own Spacebars templating language inspired by mustache/handlebars.
+To be able to utilize the form you have made in profile.html, you must create an event handler to intercept the submit event of the form in *profile.js*. Then, create a method to update the *profile* property of a user.
+
+To update this field with an object, you need to use some more 'advanced' MongoDB features than just calling insert. The update function is what you need. To update a document in a collection called *SomeObjects* with a subproperty, you can do something like this: `SomeObjects.update(id, {$set: {'subprop.somefield': 1337}});`
```

Comment:
To update a document in a collection, you can do something like this: `YourCollection.update(id, {$set: {'someField': 1337}});`